packages/odyssey-storybook/src/components/odyssey-mui/Autocomplete/Autocomplete.stories.tsx

```diff
@@ -91,54 +84,63 @@ type AutocompleteType = AutocompleteProps<
   boolean | undefined
 >;
 
-const Template: StoryFn<AutocompleteType> = (args) => {
-  return <Autocomplete {...args} options={stations} />;
-};
-
-const EmptyTemplate: StoryFn<AutocompleteType> = (args) => {
-  return <Autocomplete {...args} options={[]} />;
+export const Default: StoryObj<AutocompleteType> = {
+  render: function C(props) {
+    return <Autocomplete {...props} options={stations} />;
+  },
 };
```

Comment:
Why `function C()`? Can we avoid the single-letter name? It's confusing.

Reply:
Ah this was annoying. TLDR: eslint `react-hooks/rules-of-hooks` errors.

1. For components that just spread props to the component registered in the `StorybookMeta`, I don't specify a `render` function because Storybook automatically takes care of it, e.g., `render: (props) => <Foo {...props} />`
2. For components that **do not** use hooks, I prefer an anonymous arrow function (lambda) for brevity.
3. If the render function **does** use hooks, the eslint rule requires that the function is named and starts with a capital letter (I arbitrarily chose `C` for component) and to imply that the name itself is unimportant.
4. I favored **consistency over brevity**. The reason for using a lambda vs named function seemed an unnecessary distraction and too convoluted; later engineers will just follow whatever pattern they find in existing stories. If an engineer needs to write a render function with a hook (or adds a hook to an existing lambda) by referencing one without a hook, they'd have to go through _this_ little rabbit hole.

```tsx
// works fine
const LambdaWithoutHook: StoryObj<AutocompleteProps> = {
  render: (props) => <Autocomplete {...props} options={stations} />,
}

// eslint error  React Hook "useState" is called in function "render" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter. React Hook names must start with the word "use"  react-hooks/rules-of-hooks
const LambdaWithHook: StoryObj<AutocompleteProps> = {
  render: (props) => {
    const options = useMemo(stations, []);
    return <Autocomplete {...props} options={options} />;
  }
}

// eslint error  React Hook "useState" is called in function "render" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter. React Hook names must start with the word "use"  react-hooks/rules-of-hooks
const AnonymousFunctionWithHook: StoryObj<AutocompleteProps> = {
  render: function (props) {
    const options = useMemo(stations, []);
    return <Autocomplete {...props} options={options} />;
  }
}

// works
const NamedFunctionWithHook: StoryObj<AutocompleteProps> = {
  // any name starting with an uppercase letter will do
  render: function Foo(props) {
    const options = useMemo(stations, []);
    return <Autocomplete {...props} options={options} />;
  }
}
```

Reply:
Wow okay. Let's keep it as-is for now then. I know the Storybook folks are working on a fix as I'd actually asked them about this back in February.